src/RequestHandler.ts

```diff
@@ -105,6 +105,7 @@ async function handleFetchAsPromise<T extends HttpMethodTemplate>(
     // This error only occurs when there are network errors and such
     throw opts.errorWithCleanerStack.addContext(context).setCause(e);
   }
+  const contextIncludingBody = { ...context, body: response.statusText };
   errorContext.errorToThrow.statusCode = response.status;
```

Comment:
We never reuse this. Another reason why we don't need it. If we follow the advice above, the `statusCode` here could just be added instead to `errorContext`. eg:
```
 errorContext.statusCode = response.status
  errorContext.body = response.body
 ```